autogluon/utils/tabular/features/generators/datetime.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+import pandas as pd
+from pandas import DataFrame
+
+from .abstract import AbstractFeatureGenerator
+from ..feature_metadata import R_DATETIME, S_DATETIME_AS_OBJECT
+
+logger = logging.getLogger(__name__)
+
+
+class DatetimeFeatureGenerator(AbstractFeatureGenerator):
+    """Transforms datetime features into numeric features."""
+    def _fit_transform(self, X: DataFrame, **kwargs) -> (DataFrame, dict):
+        X_out = self._transform(X)
+        type_family_groups_special = dict(
+            datetime_as_int=list(X_out.columns)
+        )
+        return X_out, type_family_groups_special
+
+    def _transform(self, X: DataFrame) -> DataFrame:
```

Comment:
In the future, we may consider detecting during `_transform()` if at inference-time, a datetime is larger than any datetimes encountered in training-data. We could print some helpful tips in this case.

Reply:
This is an interesting idea, and is also similar to many ideas in terms of debugging. The problem is that having these checks slows down inference so we would have to be very careful about how we do them.

Reply:
Ya some sort of arg could be added to `predict()` in the future like `reliability_check=True`